packages/apalis-core/src/lib.rs

```diff
@@ -21,594 +21,78 @@
+//! ```
+//!
+//! ### WorkerBuilder
+//! [`WorkerBuilder`] provides a fluent API to configure workers with middlewares, filters, and error handling strategies.
+//!
+//! ```rust
+//! use apalis_core::worker::WorkerBuilder;
+//!
+//! let worker = WorkerBuilder::new("my-worker")
+//!     .layer(my_middleware)
+//!     .build_fn(handler);
+//! ```
+//! ### Worker
+//! A [`Worker`] represents a task processor. It polls tasks from a [`Backend`], sends them to a [`Service`], and handles retry logic, timeouts, and state transitions.
+//!
+//! Workers can be created manually or using a [`WorkerBuilder`] for more fluent composition.
+//!
+//! ### ServiceFn
+//! A [`ServiceFn`] is an adapter that turns an async function into a task [`Service`]. This is useful for injecting lightweight task handlers.
+//!
+//! Example:
+//! ```rust
+//! use apalis_core::service_fn;
+//!
+//! let handler = service_fn(|req: String, worker: WorkerContext | async move {
+//!     println!("Processing task: {:?}", req);
+//!     Ok(())
+//! });
+//! ```
+//! ### Monitor
+//! A [`Monitor`] supervises multiple [`Worker`]s, supports graceful shutdowns, and coordinates global signals like shutdown or restart
+//!
+//! Example:
+//! ```rust
+//! use apalis_core::{monitor::Monitor, worker::Worker};
+//!
+//! let worker = WorkerBuilder::new("my-worker")
+//!     .layer(my_middleware)
+//!     .build_fn(handler);
+//! Monitor::new().register(worker).run().await;
+//! ```
+//! [`Backend`]: crate::backend::Backend
+//! [`ServiceFn`]: crate::service_fn::ServiceFn
+//! [`Service`]: tower::Service
+//! [`Request<Args, Context>`]: crate::request::Request
+//! [`WorkerBuilder`]: crate::worker::builder::WorkerBuilder
+//! [`Worker`]: crate::worker::Worker
+//! [`Monitor`]: crate::monitor::Monitor
 pub mod backend;
-/// Includes all possible error types.
+/// Includes internal error types.
 pub mod error;
-/// Represents middleware offered through [`tower`]
-pub mod layers;
-/// Represents monitoring of running workers
 pub mod monitor;
-/// Represents the request to be processed.
 pub mod request;
-/// Represents different possible responses.
-pub mod response;
-/// Represents a service that is created from a function.
 pub mod service_fn;
-/// Represents ability to persist and consume jobs from storages.
-pub mod storage;
-/// Represents the utils for building workers.
 pub mod worker;
-
-/// Represents the utils needed to extend a task's context.
-pub mod data;
-/// Message queuing utilities
-pub mod mq;
-/// Allows async listening in a mpsc style.
-pub mod notify;
-/// Controlled polling and streaming
-pub mod poller;
-
-/// In-memory utilities for testing and mocking
-pub mod memory;
-
-/// Task management utilities
-pub mod task;
-
-/// Codec for handling data
-pub mod codec;
-
-/// Allows stepped tasks
-pub mod step;
-
-/// Sleep utilities
-#[cfg(feature = "sleep")]
-pub async fn sleep(duration: std::time::Duration) {
-    futures_timer::Delay::new(duration).await;
-}
```

Comment:
This is used in `monitor::Monitor::shutdown_timeout` and other crates.

Reply:
This has been refactored, I will push the latest ASAP